main_pullrequest.go

```diff
@@ -70,18 +88,30 @@ func processGitHubPullRequest(
 					pr.Sender.GetLogin(),
 				)
 			}
-			err = startPRPipeline(log, prBranchName, pr, conf, isOrgMember)
+			err = retryOnError(retryParams{
+				retryFunc: func() error {
+					return startPRPipeline(log, prBranchName, pr, conf, isOrgMember)
+				},
+				compFunc: func(err_ error) bool {
+					re := regexp.MustCompile("Missing CI config file|" +
+						"No stages / jobs for this pipeline")
+					switch {
+					case err_ == nil:
+						return false
+					case re.MatchString(err_.Error()):
+						log.Infof("start pipeline for PR '%d' is skipped", pr.Number)
+						return true
+					case !re.MatchString(err_.Error()):
+						log.Errorf("failed to start pipeline for PR: %s", err)
+						return false
```

Comment:
Aren't these swapped? :eyes: 
```suggestion
					case re.MatchString(err_.Error()):
						log.Infof("start pipeline for PR '%d' is skipped", pr.Number)
						return false
					case !re.MatchString(err_.Error()):
						log.Errorf("failed to start pipeline for PR: %s", err)
						return true
```
Otherwise I don't get it.

Reply:
No, they're not. They return the value to the comparator, and not `retry`. But I agree. This is retarded, and difficult to read. I'll switch it up in the `retry` function itself, to return a type I think, with something like a named type of retry, and not.

Thanks for the feedback, I was a little unsure of this one.

Reply:
Pushed a fix which should make this a lot clearer now

Reply:
https://github.com/mendersoftware/integration-test-runner/commit/05fb4489596fa4dcb82462cdc1c5aab56de26b90